src/vector_topic_modeling/evaluation.py

```
@@ -0,0 +1,227 @@
+    """
+    if len(clusters) < 2:
+        return {"overall_score": 0.0, "cluster_scores": {}}
+
+    cluster_vectors: list[list[list[float]]] = []
+    cluster_ids: list[str] = []
+
+    for cid, texts in clusters:
+        vecs = []
+        for text in texts:
+            if text in vectors_by_text:
+                vecs.append(vectors_by_text[text])
```

Ruff (0.15.7)

[warning] Use a list comprehension to create a transformed list

(PERF401)